zipline/assets/futures.py

```diff
@@ -13,173 +13,84 @@
-                self.root_symbol, self.as_of_date)
-        else:
-            return "FutureChain(root_symbol='%s')" % self.root_symbol
+        return "FutureChain(root_symbol='%s', as_of_date='%s')" % (
+            self.root_symbol, self.as_of_date)
 
-    def _get_datetime(self):
+    @property
+    def root_symbol(self):
         """
-        Returns the normalized simulation datetime.
+        The root symbol for this future chain.
 
         Returns
         -------
-        pandas.Timestamp
-            The normalized datetime of FutureChain's TradingAlgorithm.
+        str: The user-provided root symbol
         """
-        return normalize_date(
-            Timestamp(self._algorithm_get_datetime(), tz='UTC')
-        )
+        return self._root_symbol
 
     @property
     def as_of_date(self):
         """
-        The current as-of date of this future chain.
+        The as-of date of this future chain.
 
         Returns
         -------
-        pandas.Timestamp
-            The user-provided as_of_date if given, otherwise the
-            current datetime of the simulation.
+        pandas.Timestamp:  The user-provided as_of_date.
         """
-        if self._as_of_date is not None:
-            return self._as_of_date
-        else:
-            return self._get_datetime()
-
-    def _maybe_update_current_chain(self):
-        """ Updates the current chain if it's out of date, then returns
-            it.
+        return self._as_of_date
 
-            Returns
-            -------
-            list
-                The up-to-date current chain, a list of Future objects.
+    @property
+    def contracts(self):
         """
-        if (self._last_updated is None)\
-                or (self._last_updated != self.as_of_date):
-            self._current_chain = self._asset_finder.lookup_future_chain(
-                self.root_symbol,
-                self.as_of_date
-            )
-            self._last_updated = self.as_of_date
-
-        return self._current_chain
-
-    def __getitem__(self, key):
-        return self._maybe_update_current_chain()[key]
-
-    def __len__(self):
-        return len(self._maybe_update_current_chain())
-
-    def __iter__(self):
-        return iter(self._maybe_update_current_chain())
-
-    def as_of(self, dt):
-        """ Get the future chain for this root symbol as of a specific date.
-
-        Parameters
-        ----------
-        dt : datetime.datetime or pandas.Timestamp or str, optional
-            The as_of_date for the new chain.
-
         Returns
         -------
-        FutureChain
-
+        list: The chain of contracts for the given root symbol at the given
+        as_of_date.
         """
-        return FutureChain(
-            asset_finder=self._asset_finder,
-            get_datetime=self._algorithm_get_datetime,
-            root_symbol=self.root_symbol,
-            as_of_date=Timestamp(dt, tz='UTC'),
-        )
+        return self._contracts
```

Comment:
We probably can't give the user back a mutable reference to internal state here.  We either need to store `_contracts` as a tuple, or return a copy here.